massa-consensus-worker/src/controller.rs

```diff
@@ -219,6 +227,28 @@ impl ConsensusController for ConsensusControllerImpl {
     }
 
     fn register_block(&self, block_id: BlockId, slot: Slot, block_storage: Storage, created: bool) {
+        if self.ws_config.enabled {
+            block_storage.read_blocks().get(&block_id).map(|value| {
+                let operations: Vec<(OperationId, Option<Wrapped<Operation, OperationId>>)> = value
+                    .content
+                    .operations
+                    .iter()
+                    .map(|operation_id| {
+                        match block_storage.read_operations().get(operation_id).cloned() {
+                            Some(wrapped_operation) => (*operation_id, Some(wrapped_operation)),
+                            None => (*operation_id, None),
+                        }
+                    })
+                    .collect();
+
+                let _ = self.ws_config.block_sender.send(value.content.clone());
+                self.ws_config.filled_block_sender.send(FilledBlock {
+                    header: value.content.header.clone(),
+                    operations,
+                })
+            });
```

Comment:
You are using `map` to produce side effects, `map` should only be used when it is needed to modify the value and wield its result, please use `if let Some(value)` here